packages/ni.measurementlink.discovery.v1.client/src/ni/measurementlink/discovery/v1/client/_types.py

```diff
@@ -0,0 +1,54 @@
+"""Data types for the NI Discovery Service."""
+
+from __future__ import annotations
+
+import typing
+
+from ni.measurementlink.discovery.v1 import (
+    discovery_service_pb2,
+)
+
+
+class ServiceLocation(typing.NamedTuple):
```

Comment:
FYI, one issue with our use of NamedTuple here is that it brings a lot of baggage we might not expect, like the ability to index it by field number or use assignment to unpack parameters (e.g. `loc, port, ssl_port = x.resolve_service(...)`).

It's easy to accidentally break compatibility with these things, such as by adding new fields. However, I don't know if that justifies intentionally breaking compatibility here (such as by changing to a dataclass).